multiqc/modules/nanoq/nanoq.py

```diff
@@ -0,0 +1,303 @@
+
+        out_d = {f"{k}_{stat_type}": v for k, v in nanoq_stats.items()}
+
+        # Warn if we find overlapping data for the same sample
+        if f["s_name"] in self.nanoq_data:
+            if not set(self.nanoq_data[f["s_name"]].keys()).isdisjoint(out_d.keys()):
+                log.debug(f"Duplicate sample data found! Overwriting: {f['s_name']}")
+
+        self.nanoq_data.setdefault(f["s_name"], {}).update(out_d)
+
+        self.add_data_source(f)
+
+    def nanoq_general_stats(self) -> None:
+        """Nanoq General Stats Table"""
+        headers = {
+            "Number of reads": {
+                "title": f"# Reads ({config.long_read_count_prefix})",
+                "description": f"Number of reads ({config.long_read_count_desc})",
+                "scale": "Greens",
+                "modify": lambda x: x * config.long_read_count_multiplier,
+            },
+            "Number of bases": {
+                "title": f"Total Bases ({config.base_count_prefix})",
+                "description": f"Total bases ({config.base_count_desc})",
+                "scale": "Purples",
+                "modify": lambda x: x * config.base_count_multiplier,
+            },
+            "N50 read length": {"title": "Read N50", "description": "N50 read length", "scale": "Blues"},
+            "Longest read": {"title": "Longest Read", "description": "Longest read length (bp)", "scale": "Oranges"},
+            "Shortest read": {"title": "Shortest Read", "description": "Shortest read length (bp)", "scale": "YlOrRd"},
+            "Mean read length": {"title": "Mean Length", "description": "Mean read length (bp)", "scale": "PuBuGn"},
+            "Median read length": {
+                "title": "Median Length",
+                "description": "Median read length (bp)",
+                "scale": "RdYlBu",
+            },
+            "Mean read quality": {
+                "title": "Mean Qual",
+                "description": "Mean read quality (Phred scale)",
+                "scale": "PiYG",
+            },
+            "Median read quality": {
+                "title": "Median Qual",
+                "description": "Median read quality (Phred scale)",
+                "scale": "Spectral",
+            },
+        }
+
+        # Add columns to the general stats table
+        self.general_stats_addcols(self.nanoq_summary, headers)
+
+    def reads_by_quality_plot(self) -> None:
+        def _get_total_reads(_data_dict: Dict[str, Any]) -> Any:
+            for _stat_type in self._stat_types:
+                total_key = f"Number of reads_{_stat_type}"
+                if total_key in _data_dict:
+                    return _data_dict[total_key]
+            return None
+
+        # Get data for plot
+        data: Dict[str, Dict[str, float]] = {}
+        cats: List[str] = []
+        for name, d in self.nanoq_data.items():
+            # Iterate over available types and keep categories in same plot
+            for stat_type in self._stat_types:
```

Comment:
I don't really understand what iterating over `_stat_types` doing here. If data for multiple `_stat_types` is available, `data[name][threshold]` will be always overwritten by the latest type anyway?